### PR TITLE
Remove duplication in `toml_map_to_field` and `toml_remap`

### PR DESCRIPTION
**Pull request by TomAFrench**
# Related issue(s)

This is a very small change so an issue is overkill.

# Description

## Summary of changes

Currently we call `check_toml_map_duplicates` in each arm of the match statement in `toml_map_to_field`, the only difference between arms is in the mapped value being inserted. Instead we can just return the value to be inserted from the match statement and insert values into `field_map` in a single place below it.

IMO, this improves readability and reducing LOC is always nice. For consistency I've also changed `toml_remap` to match.

## Dependency additions / changes

N/A

## Test additions / changes

N/A

# Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [ ] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [x] I have reviewed the changes on GitHub, line by line.
- [x] I have ensured all changes are covered in the description.

# Additional context

(If applicable.)

### CODE REVIEW

File: crates/noirc_abi/src/input_parser/toml.rs
Comment (TomAFrench) [inline crates/noirc_abi/src/input_parser/toml.rs:82]: This check on the return value seems a little odd and could be removed. `BTreeMap` not having multiple values for the same key is guaranteed but I'm also not too concerned about this.

**Review by jfecher — COMMENTED** [review]
This refactoring looks good to me, but could conflict with #501. I'll wait until that PR is merged first since this is the smaller of the two.
Edit: #501 is merged, after merge conflicts are fixed with this PR we can merge it. So long as it still makes sense to do with structs with nested values.

**Review by jfecher — APPROVED** [review]
Thank you!

**Pull request merged**
